src/main/java/appeng/items/storage/AbstractStorageCell.java

```diff
@@ -268,14 +231,7 @@ private boolean disassembleDrive( final ItemStack stack, final World world, fina
 					}
 
 					// drop empty storage cell case
-					AEApi.instance().definitions().materials().emptyStorageCell().maybeStack( 1 ).ifPresent( is ->
-					{
-						final ItemStack extraA = ia.addItems( is );
-						if( !extraA.isEmpty() )
-						{
-							player.dropItem( extraA, false );
-						}
-					} );
+					dropEmptyStorageCellCase( ia, player );
 
 					if( player.inventoryContainer != null )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)